cmd/agent/main.go

```diff
@@ -1,3 +1,141 @@
 package main
 
-func main() {}
+import (
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"reflect"
+	"runtime"
+	"sync"
+	"syscall"
+	"time"
+)
+
+const (
+	requestTimeoutSeconds = 10
+)
+
+type Metrics struct {
+	runtime.MemStats
+	PollCount         uint64
+	RandomValue       uint64
+	Values            map[string]any
+	RegisteredMetrics []string
+}
+
+func NewMetrics() *Metrics {
+	return &Metrics{
+		Values: make(map[string]any),
+		RegisteredMetrics: []string{
+			"Alloc", "BuckHashSys", "Frees", "GCCPUFraction", "GCSys", "HeapAlloc", "HeapIdle", "HeapInuse", "HeapObjects",
+			"HeapReleased", "HeapSys", "LastGC", "Lookups", "MCacheInuse", "MCacheSys", "MSpanInuse", "MSpanSys", "Mallocs",
+			"NextGC", "NumForcedGC", "NumGC", "OtherSys", "PauseTotalNs", "StackInuse", "StackSys", "Sys", "TotalAlloc",
+		},
+	}
+}
+
+func (m *Metrics) collect() {
+	runtime.ReadMemStats(&m.MemStats)
+
+	m.PollCount++
+
+	var randomValue uint64
+	err := binary.Read(rand.Reader, binary.BigEndian, &randomValue)
+	if err != nil {
+		log.Fatalf("error generating random number: %v", err)
```

Comment:
```suggestion
		log.Errorf("error generating random number: %v", err)
		return err
```
Ни в коем разе не Fatalf. Агент не должен падать из-за возникновения минорных ошибок.
https://t.me/bushigo/10